torchdata/datapipes/iter/load/huggingface.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import posixpath
+
+from typing import Any, Callable, Dict, Iterator, List, Optional, Tuple, Union
+
+from torch.utils.data.datapipes.utils.common import match_masks
+
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+from torchdata.datapipes.utils import StreamWrapper
+
+try:
+    import datasets
+except ImportError:
+    datasets = None
+
+def _get_response_from_huggingface_hub(dataset, split, revision, data_files) -> Tuple[Any, StreamWrapper]:
+    dataset = datasets.load_dataset(dataset, split, revision, data_files)
+    return dataset[0], StreamWrapper(dataset)
+
+@functional_datapipe("read_from_huggingface_hub")
```

Comment:
As the first argument is a string, we can't really support `functional_datapipe` in this case.

Reply:
If the first input is not `IterDataPipe`, there won't be the option for users to call `datapipe. read_from_huggingface_hub()`

So, adding `functional_datapipe` here would cause a bug.